app/base/utils.py

```diff
@@ -2,12 +2,58 @@
 import uuid
 import numpy as np
 from datetime import datetime
+from functools import wraps
+from flask import redirect, url_for
+from itsdangerous import URLSafeTimedSerializer
+from flask_login import current_user
 from app import redis_client
 from config import S3_BUCKET_CONFIG
 
 bucket = S3_BUCKET_CONFIG["S3_BUCKET"]
 
 
+def generate_confirmation_token(user_email):
+    """
+    Generates a token and with user email as encoded data
+    """
+    serializer = URLSafeTimedSerializer(
+        os.environ.get("SECRET_KEY")
+    )
+    salt = os.environ.get("SECURITY_PASSWORD_SALT")
```

Comment:
seems like a good candidate for a util function

Reply:
I have declared them as global variables